subsys/bluetooth/host/classic/avdtp.c

```diff
@@ -30,50 +30,78 @@ LOG_MODULE_REGISTER(bt_avdtp);
 #define AVDTP_MSG_POISTION 0x00
 #define AVDTP_PKT_POSITION 0x02
 #define AVDTP_TID_POSITION 0x04
-#define AVDTP_SIGID_MASK 0x3f
+#define AVDTP_SIGID_MASK   0x3f
 
-#define AVDTP_GET_TR_ID(hdr) ((hdr & 0xf0) >> AVDTP_TID_POSITION)
+#define AVDTP_GET_TR_ID(hdr)    ((hdr & 0xf0) >> AVDTP_TID_POSITION)
 #define AVDTP_GET_MSG_TYPE(hdr) (hdr & 0x03)
 #define AVDTP_GET_PKT_TYPE(hdr) ((hdr & 0x0c) >> AVDTP_PKT_POSITION)
-#define AVDTP_GET_SIG_ID(s) (s & AVDTP_SIGID_MASK)
+#define AVDTP_GET_SIG_ID(s)     (s & AVDTP_SIGID_MASK)
 
 static struct bt_avdtp_event_cb *event_cb;
 static sys_slist_t seps;
 
 #define AVDTP_CHAN(_ch) CONTAINER_OF(_ch, struct bt_avdtp, br_chan.chan)
 
-#define AVDTP_KWORK(_work) CONTAINER_OF(CONTAINER_OF(_work, struct k_work_delayable, work),\
-					struct bt_avdtp, timeout_work)
+#define AVDTP_KWORK(_work)                                                                         \
+	CONTAINER_OF(CONTAINER_OF(_work, struct k_work_delayable, work), struct bt_avdtp,          \
+		     timeout_work)
 
 #define DISCOVER_REQ(_req) CONTAINER_OF(_req, struct bt_avdtp_discover_params, req)
-#define GET_CAP_REQ(_req) CONTAINER_OF(_req, struct bt_avdtp_get_capabilities_params, req)
+#define GET_CAP_REQ(_req)  CONTAINER_OF(_req, struct bt_avdtp_get_capabilities_params, req)
 #define SET_CONF_REQ(_req) CONTAINER_OF(_req, struct bt_avdtp_set_configuration_params, req)
-#define OPEN_REQ(_req) CONTAINER_OF(_req, struct bt_avdtp_open_params, req)
-#define START_REQ(_req) CONTAINER_OF(_req, struct bt_avdtp_start_params, req)
+#define CTRL_REQ(_req)     CONTAINER_OF(_req, struct bt_avdtp_ctrl_params, req)
 
 #define AVDTP_TIMEOUT K_SECONDS(6)
 
-K_MUTEX_DEFINE(avdtp_mutex);
-#define AVDTP_LOCK() k_mutex_lock(&avdtp_mutex, K_FOREVER)
-#define AVDTP_UNLOCK() k_mutex_unlock(&avdtp_mutex)
+K_SEM_DEFINE(avdtp_sem_lock, 1U, 1U);
 
 enum sep_state {
-	AVDTP_IDLE = 0,
-	AVDTP_CONFIGURED,
+	AVDTP_IDLE = 0x01U,
+	AVDTP_CONFIGURED = 0x02U,
 	/* establishing the transport sessions. */
-	AVDTP_OPENING,
-	AVDTP_OPEN,
-	AVDTP_STREAMING,
-	AVDTP_CLOSING,
-	AVDTP_ABORTING,
+	AVDTP_OPENING = 0x04U,
+	AVDTP_OPEN = 0x08U,
+	AVDTP_STREAMING = 0x10U,
+	AVDTP_CLOSING = 0x20U,
+	AVDTP_ABORTING = 0x40U,
```

Comment:
These look like bits to me, why not use the `BIT()` macro? Actually, why are they bits? These seem internal (not something that goes over the air) and AFAIK you can't have two states at the same time.

Reply:
yes, they are bits I will change as `BIT()`, and it can't be two states. I use it for easy `&` operation to check whether the current state is right state for the requested operation. For example: When closing the stream, the current state `AVDTP_OPEN `or `AVDTP_STREAMING` are both OK, I check it by `(param->sep->state & check_state)`.

Reply:
Updated.